### ISSUE TITLE
Suggestions for NativeScript Vue 3 Docs

### ISSUE THREAD

**Issue by neneodonkor** [nativescript-vue/nativescript-vue#1080]
Due to the fact that there is little to no information available for new users to know how to use NativeScript with Vue 3, I suggest the following should be included in the docs:

1. Environmental setup and installation. I personally had to start from the NativeScript official docs before finishing with the instructions on the NativeScript-Vue Github page.

2. A list of gotchas. Any slight differences between how Vue is used on the web vs how it is used in NativeScript, or even compared to older versions of NativeScript Vue. 

3. Code snippets or examples for workflows that our common in all apps. This could range from simple to advanced levels. Code examples could be on how to render lists, basic animations, and  how to share data from one screen to another.

4. How and when to use native code or platform specific code. 

5. I know the team is hard-pressed for time, but short tutorials can also help especially for those of us who are visual learners. Perhaps, a TODO app can be developed. You could have two versions: A simple, basic TODO with local storage. And an advanced version that should include connecting to a backend like Supabase, authentication, micro-animations, and handling of images.

6. Various ways of styling your app.

7. Performance tips.

8. How to deploy your app to Expo or Ionic's Appflow, or whatever option is out there.

9. How to use Open Native.

10. How to upgrade to a new version of NativeScript Vue.

11. How to manage or read errors. Also, common mistakes to avoid. 

12. How to enable fast refresh.

13. How to manage network activity. Creating an offline-first app.

14. App security. How to secure your APIs or store sensitive information. 

15. Finally, can we have JS examples for those of us who are not good with TS. 🫣

**Comment by vallemar** [comment]
@neneodonkor I like this, many of these points are part of NativeScript and not NativeScrip-Vue, because of this, I think it is important that anyone who goes into the documentation understands that the majority of the documentation that you should look in should be in N and only specific aspects of vue will be in the NS-vue documentation, this is because ns-vue is coupled on top of N, it is a Vue renderer that renders the N elements behind. This is why I believe that the documentation of ns-vue should be concise and only explain the parts that are really useful.

Maybe we could have a page that encompasses the points of N and that makes a redirect to the documentation of N, but I repeat, I think it is best for the developer to understand that everything related to native, views, errors and so on are part of N and not NS-Vue, although I think a page like this would be useful for many.

**Comment by neneodonkor** [comment]
@vallemar I agree with you. Perhaps, some of the suggestions could be applied to the main documentation. But then again it could also be a place on what can be done with Vue.

**Comment by rigor789** [comment]
Checked this list against the current docs. Most of it is now covered: environment setup, native and platform-specific code, styling, performance, publishing, debugging and error handling, and security are on docs.nativescript.org and apply to every flavor; installation, creating a project, navigation, the upgrade guide, devtools and plugins are on nativescript-vue.org. What is still missing is Vue-specific (a gotchas page, common snippets such as list rendering and passing data between screens, tutorials, JS examples) and belongs on the nativescript-vue.org site, so it is tracked there now rather than in this runtime repo. Closing here; thanks for the thorough list.
